sparse/numba_backend/tests/test_compressed.py

```diff
@@ -177,18 +176,21 @@ def test_tranpose(a, b):
     assert_eq(x.transpose(b), s.transpose(b))
 
 
-def test_to_scipy_sparse():
-    s = sparse.random((3, 5), density=0.5, format="gcxs", compressed_axes=(0,))
-    a = s.to_scipy_sparse()
-    b = scipy.sparse.csr_matrix(s.todense())
+@pytest.mark.parametrize("fill_value_in", [0, np.inf, np.nan, 5, None])
+@pytest.mark.parametrize("fill_value_out", [0, np.inf, np.nan, 5, None])
+@pytest.mark.parametrize("format", [sparse.COO, sparse._compressed.CSR])
+def test_to_scipy_sparse(fill_value_in, fill_value_out, format):
+    s = sparse.random((3, 5), density=0.5, format=format, fill_value=fill_value_in)
 
-    assert_eq(a, b)
+    if not ((fill_value_in in {0, None} and fill_value_out in {0, None}) or equivalent(fill_value_in, fill_value_out)):
+        with pytest.raises(ValueError):
```

Comment:
One nitpick and LGTM!

Let's make sure that error message is correct:
```
with pytest.raises(TypeError, match="{...} but should be in {...}"):
    ...
```

Reply:
Done in https://github.com/pydata/sparse/pull/685/commits/a27e4e0e80f01c49590da2b7c72c34f7ff59e1f0.